infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Action.java

```diff
@@ -0,0 +1,25 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import org.corfudb.runtime.CorfuRuntime;
+
+/**
+ *
+ * An idempotent workflow action. All workflow actions must extend this class.
```

Comment:
Why does it need to be idempotent?

Reply:
It can simplify our retry logic and the model. For example, instead of introducing a new concept like "roll back" action, the orchestrator can pick up from the last successful action.

Reply:
Seems restricting and a premature optimization. Certainly we wouldn't want all actions to follow this pattern.

Reply:
It wasn't intended as an optimization, the goal was to simplify the model. I changed the language for now.